### ISSUE TITLE
Unit of measure missing from smart meter data

### ISSUE THREAD

**Issue by frankiep** [muppet3000/homeassistant-grott#64]
Hi guys!
i'm running grott add-on by egguy and all works well, in MQTT integrations i have 2 devices, the inverter (a MIN6000TL-XH) and the smart meter (an EASTRON SDM120).
The problem is that smart meter sensors concerning power and energy have not unit of measure (you can see in screen attached) so i can not use them in Energy dashboard. Sensors about voltage, current, frequency have unit of measure.

Why? How can i solve?
Thanks
![1](https://github.com/muppet3000/homeassistant-grott/assets/3595420/def17277-9b2c-4341-8b3f-328ec4a144b5)
![sensors 1](https://github.com/muppet3000/homeassistant-grott/assets/3595420/06147938-5b51-47c8-8346-931f2722baac)
![sensors 2](https://github.com/muppet3000/homeassistant-grott/assets/3595420/911cd4a3-636b-4663-a432-66724646d60d)

**Comment by muppet3000** [comment]
That's because you're using the egguy addon and the raw mqtt values. If you want to make use of the power values on the energy dashboard you need to install this integration (the one you've logged this ticket in).

The extra thing to do is when you're installing the egguy add-on is to tweak the options show on this image: https://github.com/muppet3000/homeassistant-grott/issues/16#issuecomment-1623172859
You need to enable "Enable Grott MQTT" option and to disable the "Home Assistant plugin" option. After that my integration should then be able to be installed and make use of the data being pushed into MQTT from grott.

**Comment by frankiep** [comment]
> That's because you're using the egguy addon and the raw mqtt values. If you want to make use of the power values on the energy dashboard you need to install this integration (the one you've logged this ticket in).
> 
> The extra thing to do is when you're installing the egguy add-on is to tweak the options show on this image: [#16 (comment)](https://github.com/muppet3000/homeassistant-grott/issues/16#issuecomment-1623172859) You need to enable "Enable Grott MQTT" option and to disable the "Home Assistant plugin" option. After that my integration should then be able to be installed and make use of the data being pushed into MQTT from grott.

done but now all sensor are unknown. in your Grott integration i don't see any devices or entity :(

**Comment by muppet3000** [comment]
That's odd, it should definitely work, it will have to wait until the next publish to mqtt happens and then the data should come through. How often does your device push data to Grott?

**Comment by frankiep** [comment]
> That's odd, it should definitely work, it will have to wait until the next publish to mqtt happens and then the data should come through. How often does your device push data to Grott?

1 minute...that's something going wrong for sure

**Comment by frankiep** [comment]
and growatt service is continuing to receive data so Grott is working. There is something worng between Grott add-on and your integration

**Comment by muppet3000** [comment]
ok in which case something isn't quite right in the configuration of the mqtt pushing, what is the topic you're using? It defaults to `energy/growatt` (or something similar) and that is hard-coded in my integration currently. Is it perhaps that?

**Comment by frankiep** [comment]
> ok in which case something isn't quite right in the configuration of the mqtt pushing, what is the topic you're using? It defaults to `energy/growatt` (or something similar) and that is hard-coded in my integration currently. Is it perhaps that?

the topic is energy/grott (default for this add-on)

**Comment by muppet3000** [comment]
@egguy - Are you able to assist with this? Something isn't working quite right with @frankiep 's setup and I'm not sure where it has gone wrong.

**Comment by egguy** [comment]
I can take a look this weekend. In the meantime, did you stop the add-on then restart it manually? 
It sometimes seems the update of the configuration is not taken into account until a manual restart is issued.

**Comment by frankiep** [comment]
> I can take a look this weekend. In the meantime, did you stop the add-on then restart it manually? It sometimes seems the update of the configuration is not taken into account until a manual restart is issued.

just tried several times. Your addon is working since 1 month at least...

**Comment by frankiep** [comment]
ok maybe i got it. Stopped 2.7 version and started 2.8 beta version and now muppet3000 grott integration works!
Now i have MQTT integrations with inverter and meter devices and all their entity in unknown status, and Grott integrations with working entity. Should I disable MQTT integration to remove all that unknown entities?

**Comment by muppet3000** [comment]
> i have MQTT integrations with inverter and meter devices and all their entity in unknow

It's up to you what you do with your old entities. Glad you got it fixed.

**Comment by frankiep** [comment]
> > i have MQTT integrations with inverter and meter devices and all their entity in unknow
> 
> It's up to you what you do with your old entities. Glad you got it fixed.

ok i would not disable MQTT integration because I might need it in the future but can i disable just the entities?

than i'm taking a look to all entities from your integration...seems to be missing PV1 Energy Today from inverter. I have PV1 Energy Total, PV2 Energy Today, PV2 Energy Total, etc...

**Comment by muppet3000** [comment]
There's no reason you can't delete the old entities, but that's up to you.

It's odd that you're missing PV1 Energy Today, what type of inverter do you have? Can you give me an example of the grott log output so I can see what the name of the sensor should be?

**Comment by frankiep** [comment]
> 
> It's odd that you're missing PV1 Energy Today, what type of inverter do you have? Can you give me an example of the grott log output so I can see what the name of the sensor should be?

yes it's odd because i have PV2/PV3/PV4 Energy Today and not PV1 Energy Today...

![6](https://github.com/muppet3000/homeassistant-grott/assets/3595420/5ae893d6-88dd-4241-af21-abcbb37848f4)
![1](https://github.com/muppet3000/homeassistant-grott/assets/3595420/4eb934b5-f6a2-4678-be1f-c7bfa163cc97)
![2](https://github.com/muppet3000/homeassistant-grott/assets/3595420/096db099-3f8c-41f7-8b58-865907cf1a07)
![3](https://github.com/muppet3000/homeassistant-grott/assets/3595420/f6352535-b80c-469a-b203-26d929a95262)
![4](https://github.com/muppet3000/homeassistant-grott/assets/3595420/d599b877-9806-4cce-ae0a-70d1e2c98fcf)
![5](https://github.com/muppet3000/homeassistant-grott/assets/3595420/b71530a3-d6c4-40ec-8ac0-22b3e14b1aa8)

**Comment by frankiep** [comment]
`- MQTT jsonmsg: 
			 {"device": "SNL0CHL191", "time": "2023-11-03T17:49:52", "buffered": "no",
			 "values": {"pvserial": "SNL0CHL191", "pvstatus": 0, "pvpowerin": 0,
			 "pv1voltage": 0, "pv1current": 0, "pv1watt": 0, "pv2voltage": 0,
			 "pv2current": 0, "pv2watt": 0, "pv3voltage": 0, "pv3current": 0, "pv3watt":
			 0, "pv4voltage": 0, "pv4current": 0, "pv4watt": 0, "pvpowerout": 0,
			 "pvfrequentie": 4996, "pvgridvoltage": 2262, "pvgridcurrent": 0,
			 "pvgridpower": 0, "pvgridvoltage2": 0, "pvgridcurrent2": 0, "pvgridpower2":
			 0, "pvgridvoltage3": 0, "pvgridcurrent3": 0, "pvgridpower3": 0, "vacrs":
			 2262, "vacst": 0, "vactr": 0, "ptousertotal": 2142, "ptogridtotal": 0,
			 "ptoloadtotal": 2142, "totworktime": 14600754, "pvenergytoday": 145,
			 "pvenergytotal": 42050, "epvtotal ": 44740, **"epv1today ": 32**, "epv1total":
			 13418, "epv2today": 119, "epv2total": 31322, "epv3today": 0, "epv3total": 0,
			 "etousertoday": 20, "etousertotal": 3892, "etogridtoday": 110,
			 "etogridtotal": 28819, "eloadtoday": 61, "eloadtotal": 19813,
			 "deratingmode": 0, "iso": 65530, "dcir": 0, "dcis": 0, "dcit": 0, "gfci": 0,
			 "pvtemperature": 704, "pvipmtemperature": 704, "temp3": 0, "temp4": 0,
			 "temp5": 313, "pbusvoltage": 650, "nbusvoltage": 0, "ipf": 20000,
			 "realoppercent": 0, "opfullwatt": 0, "standbyflag": 0, "faultcode": 0,
			 "warningcode": 0, "systemfaultword0": 0, "systemfaultword1": 0,
			 "systemfaultword2": 0, "systemfaultword3": 0, "systemfaultword4": 152,
			 "systemfaultword5": 0, "systemfaultword6": 0, "systemfaultword7": 0,
			 "invstartdelaytime": 60, "bdconoffstate": 0, "drycontactstate": 0,
			 "edischrtoday": 0, "edischrtotal": 0, "echrtoday": 0, "echrtotal": 0,
			 "eacchrtoday": 0, "eacchrtotal": 0, "priority": 0, "epsfac": 0, "epsvac1":
			 0, "epsiac1": 0, "epspac1": 0, "epsvac2": 0, "epsiac2": 0, "epspac2": 0,
			 "epsvac3": 0, "epsiac3": 0, "epspac3": 0, "epspac": 0, "loadpercent": 0,
			 "pf": 10000, "dcv": 0, "bdc1_sysstatemode": 0, "bdc1_faultcode": 0,
			 "bdc1_warncode": 0, "bdc1_vbat": 0, "bdc1_ibat": 0, "bdc1_soc": 0,
			 "bdc1_vbus1": 0, "bdc1_vbus2": 0, "bdc1_ibb": 0, "bdc1_illc": 0,
			 "bdc1_tempa": 0, "bdc1_tempb": 0, "bdc1_pdischr": 0, "bdc1_pchr": 0,
			 "bdc1_edischrtotal": 0, "bdc1_echrtotal": 0, "bdc1_flag": 0,
			 "bdc2_sysstatemode": 0, "bdc2_faultcode": 0, "bdc2_warncode": 0,
			 "bdc2_vbat": 0, "bdc2_ibat": 0, "bdc2_soc": 0, "bdc2_vbus1": 0,
			 "bdc2_vbus2": 0, "bdc2_ibb": 0, "bdc2_illc": 0, "bdc2_tempa": 0,
			 "bdc2_tempb": 0, "bdc2_pdischr": 0, "bdc2_pchr": 0, "bdc2_edischrtotal": 0,
			 "bdc2_echrtotal": 0, "bdc2_flag": 0, "bms_status": 0, "bms_error": 0,
			 "bms_warninfo": 0, "bms_soc": 0, "bms_batteryvolt": 0, "bms_batterycurr": 0,
			 "bms_batterytemp": 0, "bms_maxcurr": 0, "bms_deltavolt": 0, "bms_cyclecnt":
			 0, "bms_soh": 0, "bms_constantvolt": 0, "bms_bms_info": 0, "bms_packinfo":
			 0, "bms_usingcap": 0, "bms_fw": 0, "bms_mcuversion": 0, "bms_commtype": 0}}`

i read epv1today but i do not have sensor in Grott as you can see in the screen attached in the previous reply

**Comment by muppet3000** [comment]
Looks like there's a bug in grott somewhere, looking at those logs `epvtotal` and `epv1today` have a space after them before the ending quote e.g `"epv1today "` not `"epv1today"` and therefore they're not being picked up correctly, I'd suggest submitting a bug report over in the core grott repo.
That said maybe I could see if there's something I could do on my end to handle it a bit more gracefully if this happens with other values, but I can't promise that'll happen urgently.

**Comment by frankiep** [comment]
> Looks like there's a bug in grott somewhere, looking at those logs `epvtotal` and `epv1today` have a space after them before the ending quote e.g `"epv1today "` not `"epv1today"` and therefore they're not being picked up correctly, I'd suggest submitting a bug report over in the core grott repo. That said maybe I could see if there's something I could do on my end to handle it a bit more gracefully if this happens with other values, but I can't promise that'll happen urgently.

yes! for sure the problem is there...i have to report this "error" to @egguy or to @johanmeijer?

**Comment by egguy** [comment]
I have a patch in my new version of the extension,
but the bug is in the code of the 2.8 branch: https://github.com/johanmeijer/grott/blob/Beta-(2.8.x)/grottconf.py#L1450C21-L1450C21, I think you should report to https://github.com/johanmeijer/grott/

**Comment by frankiep** [comment]
> I have a patch in my new version of the extension, The bug is in the code of the 2.8 branch: https://github.com/johanmeijer/grott/blob/Beta-(2.8.x)/grottconf.py#L1450C21-L1450C21, I think you should report to https://github.com/johanmeijer/grott/

do you have an ETA for your new version? i reported the bug to johan but without any reply..

**Comment by frankiep** [comment]
hi @egguy 
even after your 0.1.5 update the problem was not fixed...do we need to wait update from @johanmeijer?

thanks

**Comment by frankiep** [comment]
> I have a patch in my new version of the extension, The bug is in the code of the 2.8 branch: https://github.com/johanmeijer/grott/blob/Beta-(2.8.x)/grottconf.py#L1450C21-L1450C21, I think you should report to https://github.com/johanmeijer/grott/

johan finally have updated beta 2.8:

https://github.com/johanmeijer/grott/commit/c97bd8ca8c6b8dd778cd3a089711703e0a60bd7e

@egguy when do you will update your addon? thanks man!

**Comment by egguy** [comment]
I plan to make a new release but this won't be before 2024 as I'm in a zone impacted by storm, and I don't have reliable power nor internet.

**Comment by frankiep** [comment]
> Hi sorry for the late response,
> 
> I plan to make a new release but this won't be before 2024 as I'm in a zone impacted by storm, and I don't have reliable power nor internet.

okok thanks mate!

**Comment by egguy** [comment]
Hi @frankiep #, 

I have created a new version of the add-on using the 2.8.3 release. It is currently only avaialble on the -edge repo (https://github.com/egguy/ha-addons-edge/). I'm running it on my mate's HA and it seems to works perfectly.

Would you mind trying it and confirm me it's solving your problem?

**Comment by frankiep** [comment]
> Hi @frankiep,
> 
> I have created a new version of the add-on using the 2.8.3 release. It is currently only available on the -edge repo (https://github.com/egguy/ha-addons-edge/). I'm running it on my mate's HA, and it seems to work perfectly.
> 
> Would you mind trying it and confirm me it's solving your problem?

Hi @egguy 
i'm trying it just now and it seems to work good! 

PS: what is "edge" branch? it will replace the other repo?

thanks mate

**Comment by egguy** [comment]
The edge repo the "Just reased" code. it the bleeding edge version. There could be bug and other problem, but you don't need to wait for a proper release (Which I will be doing soon)

There is also a testing, who is for pre release version
